authorization-common-api/src/main/java/com/sequenceiq/authorization/resource/AlternativeAuthorizationResourceActionProvider.java

```diff
@@ -0,0 +1,30 @@
+package com.sequenceiq.authorization.resource;
+
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.ADMIN_FREEIPA;
+import static com.sequenceiq.authorization.resource.AuthorizationResourceAction.REGISTER_DATABASE;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class AlternativeAuthorizationResourceActionProvider {
```

Comment:
I don't think we need this class at all. A simple if-else will do way better in `EventAuthorizationUtils`. Currently we only have this exception where we need to check permission based on some different right.

Reply:
Please see my related comment below